pkg/storage/system_intake.go

```diff
@@ -427,3 +427,38 @@ func (s *Store) UpdateAdminLead(ctx context.Context, id uuid.UUID, adminLead str
 	)
 	return adminLead, err
 }
+
+// UpdateReviewDates updates the admin lead for an intake
+func (s *Store) UpdateReviewDates(ctx context.Context, id uuid.UUID, grbDate *time.Time, grtDate *time.Time) (uuid.UUID, error) {
+	var intake struct {
+		GRBDate   *time.Time `db:"grb_date"`
+		GRTDate   *time.Time `db:"grt_date"`
+		ID        uuid.UUID
+		UpdatedAt time.Time `db:"updated_at"`
+	}
+
+	if grbDate != nil {
+		intake.GRBDate = grbDate
+	}
+
+	if grtDate != nil {
+		intake.GRTDate = grtDate
+	}
+
+	intake.ID = id
+	intake.UpdatedAt = time.Now()
+
+	const updateSystemIntakeSQL = `
+		UPDATE system_intakes
+		SET
+			updated_at = :updated_at,
+			grb_date = :grb_date,
+			grt_date = :grt_date
+		WHERE system_intakes.id = :id
+	`
+	_, err := s.db.NamedExec(
+		updateSystemIntakeSQL,
+		intake,
+	)
+	return intake.ID, err
```

Comment:
I think this should return the system intake, just calling fetchSystemIntakeByID as the last line here. (Per previous comment) that way we can returrn the full intake to graphql, plus we get the actual values back from the database. If an error occurred silently at the database level when saving, the graphql would return potentially incorrect data